.github/workflows/tests_full.yml

```diff
@@ -12,72 +12,90 @@ jobs:
     runs-on: ubuntu-latest
     if: >-
       github.event.review.state == 'approved'
+      || github.event.review.body == 'recheck'
       || github.event.review.body == 'recheck-snap'
       || github.event.review.body == 'recheck-full'
     outputs:
-      code-changed: ${{ steps.code-changed.outputs.changed }}
-      skip: ${{ steps.skip-tests.outputs.skip }}
+      skip-func-tests: ${{ steps.skip-func-tests.outputs.skip }}
+      skip-tests: ${{ steps.skip-tests.outputs.skip }}
     steps:
       - uses: actions/checkout@v2
-        with:
-          fetch-depth: 2
-      - name: Get latest workflow run status
-        uses: actions/github-script@v4
+      - name: Get latest workflow info
         id: latest-workflow
-        with:
-          script: |
-            function sleep(seconds) {
-              return new Promise(resolve => setTimeout(resolve, seconds * 1000));
-            }
+        env:
+          CURRENT_BRANCH: ${{ github.event.pull_request.head.ref }}
+        run: |
+          function get_last_workflow() {
+            _workflows=$(curl $GITHUB_API_URL/repos/$GITHUB_REPOSITORY/actions/workflows/tests_full.yml/runs)
+            _last_workflow=$(echo "$_workflows" | jq -r --arg branch "$CURRENT_BRANCH" '.workflow_runs[1] | select(.head_branch==$branch)')
```

Comment:
Please add a comment about this. From the whole list of functional tests that have been (or are in the process of being) run, the second in the list will be checked.

If there are 5 PRs created at the same, each of them would create the previous PR that may be running func tests. So all PRs get nested waiting until the previous one is run.

This behavior makes sense.

Reply:
> If there are 5 PRs created at the same, each of them would create the previous PR that may be running func tests. So all PRs get nested waiting until the previous one is run.

No this function will only wait for running func-test in the same branch -> you couldn't create 5 PRs from same branch.

Reply:
Hmm, `curl -s https://api.github.com/repos/canonical/juju-verify/actions/workflows/tests_full.yml/runs` shows all the checks that are run via github actions. We assume that we are index0, and we look for the previous check (index1). This previous check, we check if it's part of the same PR, but it could well be from a different PR.

Imagine this use case:
1) you push a commit#1 in PR#1
2) you push a commit#1 in PR#2
3) you push another commit#2 in PR#1

The list of checks would be:
* index0: PR#1 commit#2 trying to run func tests. They run because PR#2 is related to a different .head_branch, which is not listed in index1
* index1: PR#2 commit#1 trying to run func tests. They run because PR#1 is related to a different .head_branch, which is not listed in index2 (considered "index1" for the index1 workflow of the github action)
* index2: PR#1 commit#1 trying to run func tests. They run because no previous checks are run

Reply:
Yes, That I realized at stand-up and I change it to use `jq` to select only workflows from same branch. `select(.head_branch==$branch)'` This will select only workflow from same branch <==> same PR.